### PR TITLE
Update readme for M3 milestone

### PR DESCRIPTION
**Pull request by g0g05arui**
## Summary

This PR updates the readme file in order to comply to the requirements for M3 milestone, which can be found [here](https://github.com/swent-epfl/public/blob/main/project/M3.md)

I added a technical description / overview of the app and linked the architecture diagram and figma.

### CODE REVIEW

**Comment by g0g05arui** [comment]
@1andreea26  If I have overlooked anything important or you have any suggestions regarding wording, please let me know.

File: README.md [hunk 1]
```diff
@@ -1,34 +1,115 @@
...
+
+  - ViewModel responsible for managing user data and interactions with Firebase authentication and user repository.
+  - It provides functionality to load, update, and observe user data, and to handle user-related actions such as signing in / out, adding/removing contacts, and searching users and listening for real time friend request updates.
+
+- TripViewModel:
+
+  - ViewModel for managing trip-related data and operations in the app.
+
+  - This ViewModel interacts with the `TripRepository` to fetch, create, update, and delete trips. It
+  - Handles the management of trip-related states such as the selected trip, selected day, activities, and UI state.
```
Comment (1andreea26) [inline README.md:73]: I think this should be on the above line?

File: README.md [hunk 2]
```diff
@@ -1,34 +1,115 @@
...
+##### Repositories:
+
+- Interfaces:
+
+  - `UserRepository`: Interface for managing user (basic CRUD operations) interactions with a data source.
+  - `TripRepository`: Interface for managing trip (basic CRUD operations) interactions with a data source.
+  - `PlacesRepository`: Interface for managing places-related interactions with a location (map-based) data source.
+  - `FriendRequestRepository` and `TripInviteRepository`: used managing friend requests and trip invites.
+
+- Repositproies (implementing the interfaces for Firestore and Google Maps API use cases):
```
Comment (1andreea26) [inline README.md:90]: Here the word _Repositories_ is misspelled. Make sure to change it!

File: README.md [hunk 3]
```diff
@@ -1,34 +1,115 @@
...
+- `Profile` screen: Displays user information, allows users to edit their profile, and view, accept or reject their friend requests and trips invites. Also allows users to manage their interests that may be used by the AI assistant.
+- `AddTrip` and `Settings` (for a trip) screens: Allow users to create/edit a trip and configure settings (participants).
+- `Search` screen: Allows users to search for other users and places based on location. It also displays a feed of public trips.
+- Trip screens:
+  - `WeeklyView` / `ByDay` screens: Display the trip itinerary in a weekly view or in a certain day, allowing users to view and edit activities. It also allows users to generate a schedule for a day using the AI assitant.
+  - `Map` screen: Displays a map with markers for places and activities, allowing users to view details in a trip.
+  - `Phtos` screen: Displays photos shared by participants in a trip.
+  - `Activities` screen: Displays a list of activities in a trip, allowing users to view and edit details.
```
Comment (1andreea26) [inline README.md:109]: There's also the _ActivitiesForOneDay_, the _ActivitiesMap_ Screen and the _Assistant_ Screen. We should also mention these in the README for the app.

**Comment by 1andreea26** [comment]
> @1andreea26 If I have overlooked anything important or you have any suggestions regarding wording, please let me know.

Thanks Mihai for updating the README and doing it accurately ! Looks great, there's just some screens we need to mention and some small errors.

File: README.md (hunk 3 above)
Comment (g0g05arui) [reply README.md:109]: `ActivitiesMap` is `Map screen`. I will mention the 2 others, thanks for catching this.

File: README.md (hunk 2 above)
Comment (g0g05arui) [reply README.md:90]: Good catch, I will change it asap

File: README.md (hunk 1 above)
Comment (g0g05arui) [reply README.md:73]: I will remove the "`It`" on the line above

**Review by 1andreea26 — APPROVED** [review]
Thanks for making the changes! Approved.

**Pull request merged**